src/Api/Request/AuthorizationService.php

```diff
@@ -103,6 +108,13 @@ public function createObjectsFromAuthorizationResponse(
         $assertBody = $this->assertResponseObjectCreator->createAssertObject($responseBody);
         $this->assertBuilder->createAssert($assertBody, $saferPayOrderId);
         $isPaymentSafe = $assertBody->getLiability()->getLiabilityShift();
+        $allSavedCards = array_column($this->aliasRepository->getSavedCardsNumbersByCustomerId($customerId), 'card_number');
+        $currentCard = $assertBody->getPaymentMeans()->getDisplayText();
+
+        if (in_array($currentCard, $allSavedCards)) {
```

Comment:
Hello, is this check preventing hackers to use someone else saved card ?

Reply:
Hello @joelhelvig
Thank you for asking!

This logic provides check if is current user card is saved for user. To not have same cards saved in Account page.
Answering your question, we don't have any sensitive card information, we only have last 4 numbers of card `xxxx xxxx xxxx 1234` and that's all we got from customer inputs. No sensitive data is being manipulated threw Back-end.

If you have any more questions, let us know.

Have a good day!